src/mesh/mesh_base.C

```diff
@@ -199,6 +202,10 @@ MeshBase& MeshBase::operator= (MeshBase && other_mesh)
   _node_integer_default_values = std::move(other_mesh._node_integer_default_values);
   _point_locator_close_to_point_tol = other_mesh.get_point_locator_close_to_point_tol();
 
+#ifdef LIBMESH_ENABLE_PERIODIC
+  _disconnected_boundary_pairs = std::move(other_mesh._disconnected_boundary_pairs);
```

Comment:
This should probably work for the move constructor (it's possible for users to derive a PeriodicBoundary subclass that would break here because its `clone()` kept a reference to the original mesh, but they'd have to work at it), but we need a copy in the copy constructor above too.

Reply:
Hmm... that's probably not necessary in the move constructor, but it probably is safer; I like it.